### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by edodson84** [edodson84/Aosp_Dialer_10_Gradle#9] [Ideas]
Hi this is where we can share ideas about the development of the dialer

**Comment by gigisforza70** [comment]
hi, i noticed that there are many permissions in android manifest, even device specific permissions eg samsung, huawei, oppo. Then I noticed that there are permissions for the internet and for geolocation, a general cleaning could be done. Even if there are dependencies with old and obsolete or simply useless things.

**Comment by edodson84** [comment]
those permissions are there to allow those launchers to broadcast notification dots and for shortcut access, also i plan on adding reverse lookup which is requires the internet permissions

**Comment by gigisforza70** [comment]
Ok.What about these other permissions are they really needed? Internet,camera,location,nfc

<uses-permission android:name="android.permission.INTERNET" />
<uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
<uses-permission android:name="android.permission.CAMERA" />
<uses-permission android:name="android.permission.NFC" />
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />

<meta-data
        android:name="com.google.android.backup.api_key"
        android:value="AEdPqrEAAAAIn3-Y3JKit1mrzfvcdbVhUiJn2ICtKfhGYLy0Bg" />

<uses-feature
        android:name="android.hardware.camera.any"
        android:required="false" />
